addons/token-exchange/green_secret_controller.go

```diff
@@ -108,7 +115,44 @@ func (c *greenSecretTokenExchangeAgentController) sync(ctx context.Context, sync
 		return fmt.Errorf("failed to sync hub secret %q in managed cluster in namespace %q. %v", newSecret.Name, toNamespace, err)
 	}
 
-	klog.Infof("successfully synced hub secret %q in managed clustr in namespace %q", newSecret.Name, toNamespace)
+	klog.Infof("successfully synced hub secret %q in managed cluster in namespace %q", newSecret.Name, toNamespace)
+
+	storageClusterName := string(secret.Data[common.StorageClusterNameKey])
+	err = c.updateStorageCluster(newSecret.Name, storageClusterName, toNamespace)
+	if err != nil {
+		return fmt.Errorf("failed to update secret name %q in the storageCluster %q in namespace %q. %v", newSecret.Name, storageClusterName, toNamespace, err)
+	}
+
+	return nil
+}
+
+func (c *greenSecretTokenExchangeAgentController) updateStorageCluster(secretName, storageClusterName, storageClusterNamespace string) error {
```

Comment:
Please add TODO where ever you feel code change is required to support one to many DR relationships in the future.